pymagine/vignette_filter.py

```diff
@@ -4,23 +4,74 @@
 Implementation of vignette_filter function in the pymagine package.
 """
 
-def vignette_filter(image, a=1.0):
+def vignette_filter(image_path, strength=1.0, x=0.5, y=0.5):
   """
-  Returns the given image with the vignette filter applied 
-  at the specified strength.
+  Applies vignette filter to a given image at the specified strength 
+  and focal point then saves the result to the current working directory.
   
   Parameters
   ----------
   image: string
     The local file path to the image for which the
     filter will be applied.
   
-  a: float
+  sigma: float
     Parameter for the strength of the dimming effect.
     Default: 1.0
   
-  Returns
-  -------
-  image
-    image returned with the desired filter applied
-  """
\ No newline at end of file
+  x: float
+    Parameter for the centre point of the effect
+    along the x axis.
+    Default: 0.5
+
+  y: float
+    Parameter for the centre point of the effect
+    along the y axis.
+    Default: 0.5
+  
+  """
+  if not isinstance(image_path, str):
+        raise TypeError("Image file path must be a string.")
```

Comment:
This code line also needs to be intended to the left.
```suggestion
    raise TypeError("Image file path must be a string.")
```